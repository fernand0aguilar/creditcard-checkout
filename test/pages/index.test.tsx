import React from 'react'
import { render, fireEvent } from '../testUtils'

import Index from '../../pages/index'

describe('Hello World Component', () => {
  it('matches index snapshot', () => {
    const { asFragment } = render(<Index />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('Clicking button triggers fetch function and gets response from API', () => {
    const { getByText } = render(<Index />, {})
    window.alert = jest.fn()
    new Promise(() => {
      fireEvent.click(getByText('Continuar'))
    }).then(() => {
      expect(window.alert).toHaveBeenCalledWith('data: pagar resposta ')
    })
  })
})
