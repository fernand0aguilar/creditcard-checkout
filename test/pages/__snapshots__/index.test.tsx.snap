// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello World Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <header>
      <h2>
        Header 🤖
      </h2>
    </header>
    <main>
      <h1>
        hello world 🌍
      </h1>
      <button>
        Test Button
      </button>
    </main>
    <footer>
      <h2>
        FooOooter 👣
      </h2>
    </footer>
  </div>
</DocumentFragment>
`;
